ansible/roles/test/files/tools/loganalyzer/loganalyzer_end.yml

```diff
@@ -20,8 +20,22 @@
   shell: grep "TOTAL EXPECTED MISSING MATCHES" "{{ test_out_dir }}/{{ summary_file }}" | sed -n "s/TOTAL EXPECTED MISSING MATCHES:[[:space:]]*//p"
   register: expected_missing_matches
 
+- debug: msg={{expected_missing_matches}}
+
+- name: Check if loganalyzer gets the exact number of expected messages
+  shell: grep "TOTAL EXPECTED MATCHES" "{{ test_out_dir }}/{{ summary_file }}" | sed -n "s/TOTAL EXPECTED MATCHES:[[:space:]]*//p"
+  register: expected_matches
+  when: expected_matches_target is defined
+
+- debug: msg={{expected_matches}}
```

Comment:
condition missing. when: expected_matches_target is defined

Reply:
Done. Missed some changes that are carried with other commits in this part.